test/visible_parameters_test.py

```diff
@@ -0,0 +1,95 @@
+import luigi
+from luigi.parameter import ParameterVisibility
+from helpers import unittest
+import json
+
+
+class TestTask1(luigi.Task):
+    param_one = luigi.Parameter(default='1', visibility=ParameterVisibility.HIDDEN, significant=True)
+    param_two = luigi.Parameter(default='2', significant=True)
+    param_three = luigi.Parameter(default='3', visibility=ParameterVisibility.PRIVATE, significant=True)
+
+
+class TestTask2(luigi.Task):
+    param_one = luigi.Parameter(default='1', visibility=ParameterVisibility.PRIVATE)
+    param_two = luigi.Parameter(default='2', visibility=ParameterVisibility.PRIVATE)
+    param_three = luigi.Parameter(default='3', visibility=ParameterVisibility.PRIVATE)
+
+
+class TestTask3(luigi.Task):
+    param_one = luigi.Parameter(default='1', visibility=ParameterVisibility.HIDDEN, significant=True)
+    param_two = luigi.Parameter(default='2', visibility=ParameterVisibility.HIDDEN, significant=False)
+    param_three = luigi.Parameter(default='3', visibility=ParameterVisibility.HIDDEN, significant=True)
+
+
+class TestTask4(luigi.Task):
+    param_one = luigi.Parameter(default='1', visibility=ParameterVisibility.PUBLIC, significant=True)
+    param_two = luigi.Parameter(default='2', visibility=ParameterVisibility.PUBLIC, significant=False)
+    param_three = luigi.Parameter(default='3', visibility=ParameterVisibility.PUBLIC, significant=True)
+
+
+class Test(unittest.TestCase):
+    def test_to_str_params(self):
+        task = TestTask1()
+
+        self.assertEqual(task.to_str_params(), {'param_one': '1', 'param_two': '2'})
+
+        task = TestTask2()
+
+        self.assertEqual(task.to_str_params(), {})
+
+        task = TestTask3()
+
+        self.assertEqual(task.to_str_params(), {'param_one': '1', 'param_two': '2', 'param_three': '3'})
```

Comment:
What practice do you follow to put out spaces in tests by the way? Some like the [Arrange Act Assert](http://wiki.c2.com/?ArrangeActAssert) style. We have no selected style in luigi so this is ok.

Reply:
Usually, using additional space i want to emphasize the assertion block.
Just saw this style somewhere